.github/workflows/tests.yml

```
@@ -0,0 +1,93 @@
+
+jobs:
+  track-config:
+    name: Check track configuration
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Fetch
+        run: bin/fetch-configlet
+      - name: Lint
+        run: bin/configlet lint .
+      - name: Ensure READMEs are updated
+        run: sh ./bin/ensure-readmes-are-updated.sh
+      - name: Check configlet fmt
+        run: sh ./bin/check-configlet-fmt.sh
+      - name: Install yq (for stack resolvers)
+        run: |
+         sudo add-apt-repository -y ppa:rmescandon/yq
+         sudo apt-get -q update
+         sudo apt-get -y install yq
+      - name: Ensure stack resolvers synced
+        run: sh ./bin/ensure-stack-resolvers-synced.sh
+      - name: Check for invalid UUIDs
+        # can be removed once `configlet lint` gains this ability.
+        # Check issue https://github.com/exercism/configlet/issues/99
+        run: |
+          uuids=$(jq --raw-output '.exercises | map(.uuid) | .[]' config.json)
+          bad_uuid=$(echo "$uuids" | grep -vE '^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$' || true)
+          if [ -n "$bad_uuid" ]; then
+            echo "invalid UUIDs found! please correct these to be valid UUIDs:"
+            echo "$bad_uuid"
+            exit 1
+          fi
+  exercises:
+    name: Check exercises
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        resolver:
+          - lts-from-exercises
+          - nightly
+    # Discussion seems to indicate that continue-on-error is not exactly equivalent to Travis's allow_failures, but it's the best we can do for now.
+    # If failures on nightly cause too much confusion in CI status,
+    # consider just not running against the nightly resolver.
+    # https://github.com/actions/toolkit/issues/399
+    continue-on-error: ${{ matrix.resolver == 'nightly' }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          # A little unfortunate:
+          # A change in any one exercise's package.yaml invalidates the cache for all other exercises.
+          # We are not sure there is a way around that or a better way to do things.
+          # Reason:
+          # By the https://github.com/actions/cache/blob/main/src/save.ts,
+          # the cache is not saved if it has already been generated.
+          # Thus the cache would slowly drift out of date unless we key on package.yaml contents.
+          key: ${{ matrix.resolver }}-${{ hashFiles('exercises/*/package.yaml') }}
+          path: |
+            ~/.stack
+            ~/.foldercache
+      - uses: actions/setup-haskell@v1
+        with:
+          enable-stack: true
+          stack-version: 'latest'
+      - name: Install hlint
+        run: stack --resolver ${{ matrix.resolver }} install hlint
+        if: matrix.resolver != 'lts-from-exercises'
+      - name: Install hlint
+        run: 'stack --resolver $(cat exercises/*/stack.yaml | head -1 | cut -d: -f2) install hlint'
```

Since we have yq now:

```suggestion
        run: 'stack --resolver "$(yq read "$(ls -1 exercises/*/stack.yaml | head -1)" resolver)" install hlint'
```

This is the second place we do this. Perhaps a third place warrants us to place it in a fixed location.

---

it comes at the modest cost of having to install `yq` in this `job` as well (an install in an earlier `step` in the same `job` will persist for all later steps, but installs do not persist across `job`s), but that cost is acceptable for the clarity and utility that `yq` brings (and the possibility that we want to use it elsewhere in this same `job`, so ultimately I agree it is better to take your suggestion. Coming right up.